frontend/src/pages/details-page.tsx

```diff
@@ -70,13 +94,24 @@ export default function DetailsPage() {
         <div className="relative flex flex-col">
           <img src={post.imageLink} alt={post.title} className="h-80 w-full object-cover sm:h-96" />
           <div className="absolute left-0 top-0 h-full w-full bg-slate-950/60"></div>
-          <div className="absolute top-12 w-full cursor-pointer justify-start px-2 text-lg text-slate-50 sm:top-20 sm:px-8 sm:text-xl lg:px-12 lg:text-2xl">
+          <div className="absolute top-12 flex w-full cursor-pointer justify-between px-2 text-lg text-slate-50 sm:top-20 sm:px-8 sm:text-xl lg:px-12 lg:text-2xl">
             <img
               alt="white"
               src={navigateBackWhiteIcon}
               className="active:scale-click h-5 w-10"
               onClick={() => navigate(-1)}
             />
+
+            {(post?.authorId === userData?._id || userData?.role === 'ADMIN') && (
+              <div className="flex gap-4">
```

Comment:
This is one way, but now that we have access token , please use it, instead of id

Reply:
here I am checking if the logged-in user is the author of the post or not.
I don't think it's possible to do with the access token

Reply:
@devsharmagit https://www.youtube.com/watch?v=Yh5Lil03tpI first 6mins give it a look. That's the reason we have introduced the access token, it will fail in backend because it is handled there, but i want even the frontend to have some intelligence

Reply:
@devsharmagit are you working on this? This PR looks very stale

Reply:
Can we aim to close this within couple of days, picking it

Reply:
I am not able to get the changes you are requesting here.
I am getting user data from the user state. why I use access token instead of user state?

Reply:
@siddharthisrani can you please help @devsharmagit 

@devsharmagit please sync with @siddharthisrani and @Sukomal07 and close this